meilisearch/src/routes/indexes/settings.rs

```diff
@@ -39,7 +43,7 @@ macro_rules! make_setting_route {
                 >,
                 index_uid: web::Path<String>,
             ) -> Result<HttpResponse, ResponseError> {
-                let new_settings = Settings { $attr: Setting::Reset, ..Default::default() };
+                let new_settings = Settings { $attr: Setting::Reset.into(), ..Default::default() };
```

Comment:
obviously, all these `.into()` and the duplicated `Settings` struct will need to go as well